apps/frontend/src/components/xpress/XpressProposalTable.tsx

```diff
@@ -0,0 +1,363 @@
+    });
+
+    setSearchParams((searchParam) => {
+      searchParam.delete('search');
+
+      if (searchText) searchParam.set('search', searchText);
+
+      return searchParam;
+    });
+  };
+
+  useEffect(() => {
+    if (selection && selection.length > 0) {
+      const selectionSet = new Set(selection);
+      const selected: ProposalViewData[] = [];
+      setPreselectedProposalsData(
+        proposalsData.map((proposal) => {
+          if (selectionSet.has(proposal.primaryKey.toString())) {
+            selected.push(proposal);
+          }
+
+          return {
+            ...proposal,
+            tableData: {
+              checked: selectionSet.has(proposal.primaryKey.toString()),
+            },
+          };
+        })
+      );
+
+      setSelectedProposals(selected);
+    } else {
+      setPreselectedProposalsData(
+        proposalsData.map((proposal) => ({
+          ...proposal,
+          tableData: { checked: false },
+        }))
+      );
+      setSelectedProposals([]);
+    }
+  }, [proposalsData, JSON.stringify(selection)]);
+
+  const handleSortOrderChange = (orderByCollection: OrderByCollection[]) => {
+    const [orderBy] = orderByCollection;
+    setSearchParams((searchParam) => {
+      searchParam.delete('sortField');
+      searchParam.delete('sortDirection');
+      if (orderBy?.orderByField != null && orderBy?.orderDirection != null) {
+        searchParam.set('sortField', orderBy?.orderByField);
+        searchParam.set('sortDirection', orderBy?.orderDirection);
+      }
+
+      return searchParam;
+    });
+  };
+
+  useEffect(() => {
+    setPreselectedProposalsData(proposalsData);
+  }, [proposalsData, queryParameters]);
+
+  return (
+    <>
+      <StyledContainer maxWidth={false}>
+        <StyledPaper data-cy="xpress-proposals-table">
+          <XpressProposalFilterBar
+            calls={{ data: calls, isLoading: loadingCalls }}
+            instruments={{ data: instruments, isLoading: loadingInstruments }}
+            techniques={{
+              data: techniques,
+              isLoading: loadingTechniques,
+            }}
+            proposalStatuses={{
+              data: proposalStatuses,
+              isLoading: loadingProposalStatuses,
+            }}
+            setProposalFilter={setProposalFilter}
+            filter={proposalFilter}
+          />
+          <MaterialTable<ProposalViewData>
+            icons={tableIcons}
+            title={'Xpress Proposals'}
+            columns={columns}
+            data={tableData}
+            totalCount={20}
+            isLoading={loading}
+            options={{
+              search: true,
+              searchText: search || undefined,
+              selection: true,
+              headerSelectionProps: {
+                inputProps: { 'aria-label': 'Select All Rows' },
+              },
+              debounceInterval: 600,
+              columnsButton: true,
+              pageSize: 20,
```

Comment:
Are we not allowing page size changes here.

Reply:
20 which is set here will be the default size. We do can change the page size/the number of rows from UI.

Reply:
How are we then updating it when the use select a new page size.